Week04/lesson08/homework/temp_converter/js/indexbonus4.js

```diff
@@ -0,0 +1,153 @@
+  var f = document.querySelector("#f").value;
+
+  if ((f === null || f === "") && (c === null || c === "")) {
+      document.querySelector("#alert").innerHTML = "Please fill one of the fields above.";
+      document.querySelector("#alert").style.border = "solid 1px red";
+      document.querySelector("#alert").style.backgroundColor = "white";
+      document.querySelector("#alert").style.padding = "5px";
+      document.querySelector("#c").style.borderColor = "red";
+      document.querySelector("#f").style.borderColor = "red";
+  }
+
+  else if ((f === null || f === "") && (c < 0)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "160px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://blogldc.s3.amazonaws.com/wp-content/uploads/2015/01/2015_01_16_coldphotos1.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((f === null || f === "") && (c < 10)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "120px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((f === null || f === "") && (c < 20)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "80px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((f === null || f === "") && (c < 30)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "40px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if (f === null || f === "") {
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "0px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://img.fotocommunity.com/eruption-0076bbc1-8b12-41ef-99ce-f4ea39b80229.jpg?height=400')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((c === null || c === "") & (f < 32)) {
```

Comment:
I'm sure you've realised this now that you've seen the solution in class, but basically here you're repeating yourself because ultimately `c < 0` is the same thing a `f < 32`. So instead I recommend breaking up the if/else that determines the conversion from the if/else that determines the styling. So based on your existing code, that would look something like this:

```
if (fNotFilledIn && cNotFilledIn) { /* neither filled in */
    document.querySelector("#alert").innerHTML = "Please fill one of the fields above.";
  } else if (fNotFilledIn) {
    f = c * 9 / 5 + 32;
    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
    document.querySelector("#c").value = c + "°C";
  } else if (cNotFilledIn) {
    c = (f - 32) * 5 / 9;
    document.querySelector("#c").value = Math.round(c*10)/10+"°C";
    document.querySelector("#c").value = f + "°F";
  } else { /* both filled in */
    document.querySelector("#alert").innerHTML = "Please only fill <span id='emphasize'>one</span> of the fields above. Click 'Reset' to clear both fields.";
  }

  if (fNotFilledIn && cNotFilledIn) { /* neither filled in */
    document.querySelector("#alert").style.border = "solid 1px red";
    document.querySelector("#alert").style.backgroundColor = "white";
    document.querySelector("#alert").style.padding = "5px";
    document.querySelector("#c").style.borderColor = "red";
    document.querySelector("#f").style.borderColor = "red";
  } else if (!fNotFilledIn && !cNotFilledIn) { /* both filled in */
    document.querySelector("#alert").style.borderTop = "solid 1px red";
    document.querySelector("#alert").style.borderBottom = "solid 1px red";
    document.querySelector("#alert").style.backgroundColor = "white";
    document.querySelector("#alert").style.padding = "5px";
    document.querySelector("#c").style.borderColor = "red";
    document.querySelector("#f").style.borderColor = "red";
  } else if (c < 0) {
    document.querySelector("#thermo").style.paddingTop = "160px";
    document.querySelector(".wrapper").style.backgroundImage = "url('http://blogldc.s3.amazonaws.com/wp-content/uploads/2015/01/2015_01_16_coldphotos1.jpg')";
    document.querySelector(".wrapper").style.backgroundSize = "cover";
  } else if (c < 10) {
    document.querySelector("#thermo").style.paddingTop = "120px";
    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
    document.querySelector(".wrapper").style.backgroundSize = "cover";
  } else if (c < 20) {
    document.querySelector("#thermo").style.paddingTop = "80px";
    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
    document.querySelector(".wrapper").style.backgroundSize = "cover";
  } else if (c < 30) {
    document.querySelector("#thermo").style.paddingTop = "40px";
    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
    document.querySelector(".wrapper").style.backgroundSize = "cover";
  } else { /* c is 30 or greater */
    document.querySelector("#thermo").style.paddingTop = "0px";
    document.querySelector(".wrapper").style.backgroundImage = "url('http://img.fotocommunity.com/eruption-0076bbc1-8b12-41ef-99ce-f4ea39b80229.jpg?height=400')";
    document.querySelector(".wrapper").style.backgroundSize = "cover";
  }
```

... obviously there's loads here you could still clean up, taking advantage of functions to do some of the repetitive code.